src/components/TrapFocus/tests/TrapFocus.test.tsx

```diff
@@ -92,21 +96,47 @@ describe('<TrapFocus />', () => {
       .find('button')
       .getDOMNode();
 
-    it('prevents default when focus moves to an external node', () => {
-      const trapFocus = mountWithAppProvider(
-        <TrapFocus>
-          <TextField label="" value="" onChange={noop} autoFocus />
-        </TrapFocus>,
-      );
+    const trapFocus = mountWithAppProvider(
+      <TrapFocus>
+        <TextField label="" value="" onChange={noop} autoFocus />
+        <TextField label="" value="" onChange={noop} autoFocus />
+      </TrapFocus>,
+    );
 
-      const event: FocusEvent = new FocusEvent('focusout', {
-        relatedTarget: externalDomNode,
+    const event: FocusEvent = new FocusEvent('focusout', {
+      relatedTarget: externalDomNode,
+    });
+    Object.assign(event, {preventDefault: jest.fn()});
+
+    describe('prevents default when focus moves to an external node', () => {
+      let rafSpy: jest.SpyInstance;
+
+      beforeEach(() => {
+        rafSpy = jest.spyOn(window, 'requestAnimationFrame');
+        rafSpy.mockImplementation((callback) => callback());
       });
-      Object.assign(event, {preventDefault: jest.fn()});
 
-      trigger(trapFocus.find(EventListener), 'handler', event);
+      afterEach(() => {
+        rafSpy.mockRestore();
+      });
+
+      it('has one focusable node', () => {
+        trigger(trapFocus.find(EventListener), 'handler', {
+          ...event,
+          srcElement: findFirstFocusableNode(trapFocus.getDOMNode()),
+        });
+
+        expect(event.preventDefault).toHaveBeenCalled();
+      });
+
+      it('it has multiple focusable nodes', () => {
+        trigger(trapFocus.find(EventListener), 'handler', {
+          ...event,
+          srcElement: findLastFocusableNode(trapFocus.getDOMNode()),
```

Comment:
If there are **multiple** focusable nodes in a container, then the first and second element are _not_ the same so `srcElement` will be the last element in this test.